capstone/src/main/webapp/script.js

```diff
@@ -12,44 +12,219 @@
+  // Draw the separating line for the output
+  const separator = document.getElementById('separator-container');
+  separator.innerHTML = '';
+  separator.appendChild(document.createElement('hr'));
+
+  // Get detailed analysis if requested
+  document.getElementById('analysis-container').innerHTML = '';
+  const radios = document.getElementsByName('analysisRadios');
+  var delimiter = null;
+  for (i = 0; i < radios.length; i++) {
+    if (radios[i].checked) {
+      delimiter = radios[i].value;
+      break;
+    }
+  }
+  if (delimiter != "") {
+    getAnalysis(textElement.value, langElement.value, requestedAttributes, delimiter); 
+  }
+}
 
-  if (toxicityData.attributeScores) {
-    for (let key in toxicityData.attributeScores) {
-      if (toxicityData.attributeScores[key].summaryScore && 
-          toxicityData.attributeScores[key].summaryScore.value) {
-        attributeElement = createAnyElement('p', key + ": " + toxicityData.attributeScores[key].summaryScore.value);
-        outputElement.appendChild(attributeElement);
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, requestedAttributes, delimiter) {
+  // Set up the detailed analysis section
+  const analysisContainer = document.getElementById('analysis-container')
+  analysisContainer.appendChild(createAnyElement('b', 'Detailed Anaylsis'));
+  const result = createAnyElement('p', '');
+  result.className = 'detailed-analysis';
+  
+  // Generate the results for every substring of the input text
+  const substrings = text.split(delimiter);
+  for (i = 0; i < substrings.length; i++) {
+    if (substrings[i] != '') {
+      // Get the Perspective scores for the substring & sort them descending
+      const response = await callPerspective(substrings[i], lang, requestedAttributes);
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
+      var attributes = [];
+      Object.keys(response.attributeScores).forEach((attribute) => {
+        attributes.push([attribute, response.attributeScores[attribute].summaryScore.value]);
+      });
+      attributes.sort(function(a, b) {
+        return b[1] - a[1];
+      });
+
+      // Color the substring appropriately	
+      const substringEl = createAnyElement('span', substrings[i] + delimiter);
+      substringEl.className = 'green-background segment';
+      if (toxicityScore >= 0.8) {
+        substringEl.className = 'red-background segment';
+      } else if (toxicityScore >= 0.2) {
+        substringEl.className = 'yellow-background segment';
       }
+     
+      // Attach a tooltip (info-box for the substring)
+      const tooltipEl = createTooltip(attributes);
+      substringEl.appendChild(tooltipEl)
+      result.appendChild(substringEl);
+
     }
   }
+  analysisContainer.appendChild(result);
+}
+
+/** Creates a tooltip for a substring in the detailed analysis */
+function createTooltip(attributes) {
+  const tooltipEL = document.createElement('div');
```

Comment:
nit:
```suggestion
  const tooltipEl = document.createElement('div');
```

Reply:
This would also require changing the other instances of `tooltipEL` to `tooltipEl`

Reply:
Fixed